File: src/api/agentAccess.ts
import type { AgentGuideStatusPayload } from "../bindings/AgentGuideStatusPayload";
import type { AgentId } from "../bindings/AgentId";
import { invokeCommand } from "./transport";

export const agentAccessApi = {
  getStatuses(): Promise<AgentGuideStatusPayload[]> {
    return invokeCommand("get_agent_guide_statuses");
  },

  install(agents: AgentId[]): Promise<AgentGuideStatusPayload[]> {
    return invokeCommand("install_agent_guides", { agents });
  },

  remove(agent: AgentId, removeSharedMemory: boolean): Promise<AgentGuideStatusPayload[]> {
    return invokeCommand("remove_agent_guide", { agent, removeSharedMemory });
  },

  reveal(agent: AgentId): Promise<void> {
    return invokeCommand("reveal_agent_guide", { agent });
  },
};

File: src-tauri/src/commands/agent_access.rs
use tauri::{AppHandle, State};

use crate::{
    agent_access::{AgentId, GuideLifecycle},
    ipc::AgentGuideStatusPayload,
    platform,
};

#[tauri::command]
pub(crate) fn get_agent_guide_statuses(
    lifecycle: State<'_, GuideLifecycle>,
) -> Result<Vec<AgentGuideStatusPayload>, String> {
    lifecycle
        .statuses()
        .map(|statuses| statuses.into_iter().map(Into::into).collect())
}

#[tauri::command]
pub(crate) fn install_agent_guides(
    lifecycle: State<'_, GuideLifecycle>,
    agents: Vec<AgentId>,
) -> Result<Vec<AgentGuideStatusPayload>, String> {
    lifecycle
        .install_guides(&agents)
        .map(|statuses| statuses.into_iter().map(Into::into).collect())
}

#[tauri::command]
pub(crate) fn remove_agent_guide(
    lifecycle: State<'_, GuideLifecycle>,
    agent: AgentId,
    remove_shared_memory: bool,
) -> Result<Vec<AgentGuideStatusPayload>, String> {
    lifecycle
        .remove_guide(agent, remove_shared_memory)
        .map(|statuses| statuses.into_iter().map(Into::into).collect())
}

#[tauri::command]
pub(crate) fn reveal_agent_guide(
    app: AppHandle,
    lifecycle: State<'_, GuideLifecycle>,
    agent: AgentId,
) -> Result<(), String> {
    platform::reveal_in_finder(&app, &lifecycle.guide_dir(agent))
}
